Standardize kube-dns policy examples to use service match syntax (#2588)
tigera/docs: calico_versioned_docs/version-3.31/network-policy/policy-rules/service-policy.mdx

:::

:::tip

> You may see examples elsewhere that use a label selector (for example, `selector: 'k8s-app == "kube-dns"'`) with explicit protocol and port rules to allow DNS traffic. That approach can be useful when you need to target pods directly without a backing Kubernetes Service, but the service match syntax shown above is preferred because it is simpler and automatically resolves the correct endpoints and ports.

Author: 🤖 Good point — updated the tip to also mention restricting to a subset of a Service's ports/protocols as a reason to use the label selector approach. Fixed in 51bfb19.

> You may see examples elsewhere that use a label selector (for example, `selector: 'k8s-app == "kube-dns"'`) with explicit protocol and port rules to allow DNS traffic. That approach can be useful when you need to target pods directly without a backing Kubernetes Service, or when you need to restrict traffic to only a subset of a Service's ports or protocols. The service match syntax shown above is preferred in the common case because it is simpler and automatically resolves the correct endpoints and ports.

Author: 🤖 **Style suggestion** (via `docs-style-checker` skill)

This tip block has three style items:

1. **Hedging** — "You may see examples elsewhere" is a soft opening. Lead with the point instead.
2. **Long sentence** — "That approach can be useful when..." is 34 words with two distinct use cases packed in.
3. **Passive + directional ref** — "shown above is preferred" — passive voice, and "above" is a positional reference that can break in different layouts.

Suggested rewrite (applies to all 3 versioned copies):

```suggestion
Some examples use a label selector (for example, `selector: 'k8s-app == "kube-dns"'`) with explicit protocol and port rules to allow DNS traffic. That approach works when you need to target pods directly without a backing Kubernetes Service, or restrict traffic to a subset of a Service's ports or protocols. Use the service match syntax for the common case — it is simpler and automatically resolves the correct endpoints and ports.
```
